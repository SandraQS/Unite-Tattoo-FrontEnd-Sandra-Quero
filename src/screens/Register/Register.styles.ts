import { StyleSheet } from "react-native";
import { uniteTattoo } from "../../styles/uniteTatto.styles";

const { color, fontSize } = uniteTattoo;
export default StyleSheet.create({
  checkbox: {},

  forms: {
    marginTop: 30,
  },

  input: {
    borderBottomColor: color.black,
    borderBottomWidth: 1,
    padding: 20,
  },
  mainContainer: {
    backgroundColor: color.mediumBrown,
    padding: 20,
  },
  text: {
    fontSize: fontSize.text,
    textAlign: "center",
  },

  tittleH2: {
    fontSize: fontSize.tittleH2,
    textAlign: "center",
  },
  welcomeContainer: {
    alignContent: "center",
    backgroundColor: color.lightBrown,
    flex: 1,
    padding: 30,
  },
});
