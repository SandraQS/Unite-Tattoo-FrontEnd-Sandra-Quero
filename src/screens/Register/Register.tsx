import { useNavigation } from "@react-navigation/core";
import React, { useEffect, useState } from "react";
import { colors } from "../../styles/uniteTatto.styles";

import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  Text,
  View,
  TextInput,
} from "react-native";
import CheckBox from "@react-native-community/checkbox";
import GeneralButton from "../../components/GeneralButton/GeneralButton";
import { LoginScreenNavigationProp } from "../../types/navigation.types";

import styles from "./Register.styles";
import { useTattooArtist } from "../../hooks/useTattooArtist";
import AutoHeightImage from "react-native-auto-height-image";

const Register = () => {
  const navigation = useNavigation<LoginScreenNavigationProp>();
  const { tattooArtistCreate } = useTattooArtist();

  const initTatooArtisData = {
    personalDataTattoArtist: {
      name: "",
      surname1: "",
      surname2: "",
    },

    userDataTattoArtist: {
      userName: "",
      password: "",
      repeatPassword: "",
      email: "",
    },

    professionalDataTattooArtist: {
      studioName: "",
      professionalName: "",
      phone: "",
      contactEmail: "",
      openingHours: "",
      direction: "",
      tattooStyles: "",
      colaboration: false,
    },
  };

  const [personalDataTattoArtist, setPersonalDataTattoArtist] = useState(
    initTatooArtisData.personalDataTattoArtist
  );
  const [userDataTattoArtist, setUserDataTattoArtist] = useState(
    initTatooArtisData.userDataTattoArtist
  );
  const [professionalDataTattooArtist, setProfessionalDataTattooArtist] =
    useState(initTatooArtisData.professionalDataTattooArtist);

  const [isSelected, setSelection] = useState(false);
  const [newTattooArtist, setNewTattooArtist] = useState(initTatooArtisData);

  const onChangeDataPersonal = (event: any) => {
    event.persist();

    const { text } = event.nativeEvent;
    const { testID } =
      event.target._internalFiberInstanceHandleDEV.return.child.pendingProps;

    setPersonalDataTattoArtist({
      ...personalDataTattoArtist,
      [testID]: text,
    });
    return personalDataTattoArtist;
  };

  const onChangeDataUser = (event: any) => {
    event.persist();

    const { text } = event.nativeEvent;
    const { testID } =
      event.target._internalFiberInstanceHandleDEV.return.child.pendingProps;

    setUserDataTattoArtist({
      ...userDataTattoArtist,
      [testID]: text,
    });
    return userDataTattoArtist;
  };

  const onChangeDataProfessional = (event: any) => {
    event.persist();

    const { text } = event.nativeEvent;
    const { testID } =
      event.target._internalFiberInstanceHandleDEV.return.child.pendingProps;

    setProfessionalDataTattooArtist({
      ...professionalDataTattooArtist,
      [testID]: text,
    });
    return professionalDataTattooArtist;
  };

  const selectCheckBox = () => {
    setSelection(!isSelected);
    setProfessionalDataTattooArtist({
      ...professionalDataTattooArtist,
      colaboration: isSelected,
    });
  };

  useEffect(() => {
    setNewTattooArtist({
      personalDataTattoArtist: {
        ...personalDataTattoArtist,
      },

      userDataTattoArtist: {
        ...userDataTattoArtist,
      },

      professionalDataTattooArtist: {
        ...professionalDataTattooArtist,
      },
    });
  }, [
    personalDataTattoArtist,
    userDataTattoArtist,
    professionalDataTattooArtist,
  ]);

  const onSubmit = () => {
    tattooArtistCreate(newTattooArtist);
    setPersonalDataTattoArtist(initTatooArtisData.personalDataTattoArtist);
    setUserDataTattoArtist(initTatooArtisData.userDataTattoArtist);
    setProfessionalDataTattooArtist(
      initTatooArtisData.professionalDataTattooArtist
    );
    setSelection(false);
    setNewTattooArtist(initTatooArtisData);
  };

  return (
    <SafeAreaView>
      <StatusBar />
      <ScrollView>
        <View style={styles.mainContainer}>
          <View style={styles.logoContainter}>
            <AutoHeightImage
              width={150}
              source={require("../../assets/utoo-logo.png")}
              style={styles.logo}
            />
          </View>
          <View style={styles.welcomeContainer}>
            <Text style={styles.tittleH2}>Bienvenid@</Text>
            <View>
              <Text style={styles.text}>
                Has escogido la opción “tatuador”, a continuación deberás
                rellenar un formulario para registrarte y así poder formar parte
                de nuestra comunidad. Después de eso, podrás entrar en tu perfil
                como tatuador directamente. Esperamos que tengas una buena
                experiencia!
              </Text>
            </View>
          </View>

          <View style={styles.forms}>
            <Text style={styles.tittleH2}>Datos personales</Text>
            <View>
              <View>
                <TextInput
                  style={styles.input}
                  value={personalDataTattoArtist.name}
                  // onFocus={}
                  placeholder="Nombre *"
                  onChange={onChangeDataPersonal}
                  testID="name"
                  maxLength={20}
                />
              </View>

              <View>
                <TextInput
                  style={styles.input}
                  value={personalDataTattoArtist.surname1}
                  onChange={onChangeDataPersonal}
                  // onFocus={}
                  placeholder="Primer apellido *"
                  testID="surname1"
                  maxLength={20}
                />
              </View>
              <View>
                <TextInput
                  style={styles.input}
                  value={personalDataTattoArtist.surname2}
                  onChange={onChangeDataPersonal}
                  // onFocus={}
                  placeholder="Segundo apellido *"
                  testID="surname2"
                  maxLength={20}
                />
              </View>
            </View>
          </View>

          <View style={styles.forms}>
            <Text style={styles.tittleH2}>Datos de usuario</Text>
            <View>
              <View>
                <TextInput
                  style={styles.input}
                  value={userDataTattoArtist.userName}
                  // onFocus={}
                  placeholder="Nombre de usuario *"
                  onChange={onChangeDataUser}
                  testID="userName"
                  maxLength={20}
                />
              </View>

              <View>
                <TextInput
                  style={styles.input}
                  value={userDataTattoArtist.password}
                  onChange={onChangeDataUser}
                  // onFocus={}
                  placeholder="Contraseña *"
                  testID="password"
                  secureTextEntry={true}
                  maxLength={20}
                />
              </View>
              <View>
                <TextInput
                  style={styles.input}
                  value={userDataTattoArtist.repeatPassword}
                  onChange={onChangeDataUser}
                  // onFocus={}
                  placeholder="Repetir la contraseña *"
                  secureTextEntry={true}
                  testID="repeatPassword"
                  maxLength={20}
                />
              </View>

              <View>
                <TextInput
                  keyboardType="email-address"
                  style={styles.input}
                  value={userDataTattoArtist.email}
                  onChange={onChangeDataUser}
                  // onFocus={}
                  placeholder="Email *"
                  testID="email"
                  maxLength={20}
                />
              </View>
            </View>
          </View>

          <View style={styles.forms}>
            <Text style={styles.tittleH2}>Datos profesionales</Text>
            <View>
              <View>
                <TextInput
                  style={styles.input}
                  value={professionalDataTattooArtist.studioName}
                  // onFocus={}
                  placeholder="Nombre Estudio (opcional)"
                  onChange={onChangeDataProfessional}
                  testID="studioName"
                  maxLength={20}
                />
              </View>

              <View>
                <TextInput
                  style={styles.input}
                  value={professionalDataTattooArtist.professionalName}
                  // onFocus={}
                  placeholder="Nombre profesional *"
                  onChange={onChangeDataProfessional}
                  testID="professionalName"
                  maxLength={20}
                />
              </View>

              <View>
                <TextInput
                  keyboardType="numeric"
                  style={styles.input}
                  value={professionalDataTattooArtist.phone}
                  // onFocus={}
                  placeholder="Telefono de contacto *"
                  onChange={onChangeDataProfessional}
                  testID="phone"
                  maxLength={20}
                />
              </View>

              <View>
                <TextInput
                  keyboardType="email-address"
                  style={styles.input}
                  value={professionalDataTattooArtist.contactEmail}
                  // onFocus={}
                  placeholder="Email de contacto *"
                  onChange={onChangeDataProfessional}
                  testID="contactEmail"
                  maxLength={20}
                />
              </View>
            </View>

            <View>
              <TextInput
                style={styles.input}
                value={professionalDataTattooArtist.openingHours}
                // onFocus={}
                placeholder="Horario *"
                onChange={onChangeDataProfessional}
                testID="openingHours"
                maxLength={20}
              />
            </View>

            <View>
              <TextInput
                style={styles.input}
                value={professionalDataTattooArtist.direction}
                // onFocus={}
                placeholder="Dirección *"
                onChange={onChangeDataProfessional}
                testID="direction"
                maxLength={20}
              />
            </View>

            <View>
              <TextInput
                style={styles.input}
                value={professionalDataTattooArtist.tattooStyles}
                // onFocus={}
                placeholder="Estilos *"
                onChange={onChangeDataProfessional}
                testID="tattooStyles"
                maxLength={20}
              />
            </View>
          </View>

          <View>
            <CheckBox
              disabled={false}
              value={professionalDataTattooArtist.colaboration}
              onValueChange={selectCheckBox}
              boxType="square"
              testID="colaboration"
              onTintColor={colors.darkBrown}
              onCheckColor={colors.darkBrown}
              tintColor={colors.black}
            />
            <Text style={styles.text}>
              Colaboración con otros estudios / tatuadores?
            </Text>

            <GeneralButton
              textButton="REGISTRARME COMO TATUADOR"
              functionOnPress={onSubmit}
            />
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

export default Register;
