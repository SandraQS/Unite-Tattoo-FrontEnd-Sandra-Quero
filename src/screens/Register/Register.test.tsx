import { render } from "@testing-library/react-native";
import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import configureStore from "../../redux/store";
import { Provider } from "react-redux";
import Register from "./Register";

describe("Given a Register component", () => {
  describe("When it is rendered", () => {
    test("Then it should show match the last snapshot", () => {
      const store = configureStore();
      const screen = render(
        <Provider store={store}>
          <NavigationContainer>
            <Register />
          </NavigationContainer>
        </Provider>
      );

      expect(screen).toMatchSnapshot();
    });
  });
});
