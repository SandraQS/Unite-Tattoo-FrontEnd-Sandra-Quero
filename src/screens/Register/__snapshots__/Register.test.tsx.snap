// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Register component When it is rendered Then it should show match the last snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#A19178",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "marginTop": 80,
            "padding": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Image
            onError={[Function]}
            source={
              Object {
                "testUri": "../../../src/assets/utoo-logo.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 0,
                  "width": 150,
                },
                Object {
                  "margin": 20,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "#DFD8D2",
              "flex": 1,
              "padding": 30,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 25,
                "textAlign": "center",
              }
            }
          >
            Bienvenid@!
          </Text>
          <View>
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "textAlign": "center",
                }
              }
            >
              Has escogido la opción
              <Text
                style={
                  Object {
                    "fontWeight": "800",
                  }
                }
              >
                 “tatuador”,
              </Text>
               a continuación deberás rellenar un formulario para registrarte y así poder formar parte de nuestra comunidad. Después de eso, podrás entrar en tu perfil como tatuador directamente. Esperamos que tengas una buena experiencia!
            </Text>
          </View>
          <View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "opacity": 0.5,
                  "paddingTop": 10,
                }
              }
            >
              <Image
                onError={[Function]}
                source={
                  Object {
                    "testUri": "../../../src/assets/icono-rombo-titulo.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 0,
                      "width": 30,
                    },
                    undefined,
                  ]
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 18,
                      "textAlign": "center",
                    },
                    Object {
                      "marginLeft": 5,
                    },
                  ]
                }
              >
                Cancelar
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 30,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginVertical": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                }
              }
            >
              <Image
                onError={[Function]}
                source={
                  Object {
                    "testUri": "../../../src/assets/icono-rombo-titulo.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 0,
                      "width": 50,
                    },
                    undefined,
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 3,
                  "left": 30,
                  "position": "absolute",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontSize": 20,
                  }
                }
              >
                DATOS PERSONALES
                <Text
                  style={
                    Object {
                      "fontWeight": "800",
                    }
                  }
                />
              </Text>
            </View>
          </View>
          <View>
            <View>
              <TextInput
                maxLength={20}
                onChange={[Function]}
                placeholder="Nombre *"
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="name"
                value=""
              />
            </View>
            <View>
              <TextInput
                maxLength={20}
                onChange={[Function]}
                placeholder="Primer apellido *"
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="surname1"
                value=""
              />
            </View>
            <View>
              <TextInput
                maxLength={20}
                onChange={[Function]}
                placeholder="Segundo apellido *"
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="surname2"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 30,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginVertical": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                }
              }
            >
              <Image
                onError={[Function]}
                source={
                  Object {
                    "testUri": "../../../src/assets/icono-rombo-titulo.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 0,
                      "width": 50,
                    },
                    undefined,
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 3,
                  "left": 30,
                  "position": "absolute",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontSize": 20,
                  }
                }
              >
                DATOS DE USUARIO
                <Text
                  style={
                    Object {
                      "fontWeight": "800",
                    }
                  }
                />
              </Text>
            </View>
          </View>
          <View>
            <View>
              <TextInput
                maxLength={20}
                onChange={[Function]}
                placeholder="Nombre de usuario *"
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="userName"
                value=""
              />
            </View>
            <View>
              <TextInput
                maxLength={20}
                onChange={[Function]}
                placeholder="Contraseña *"
                secureTextEntry={true}
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="password"
                value=""
              />
            </View>
            <View>
              <TextInput
                maxLength={20}
                onChange={[Function]}
                placeholder="Repetir la contraseña *"
                secureTextEntry={true}
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="repeatPassword"
                value=""
              />
            </View>
            <View>
              <TextInput
                keyboardType="email-address"
                maxLength={20}
                onChange={[Function]}
                placeholder="Email *"
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="email"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 30,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginVertical": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                }
              }
            >
              <Image
                onError={[Function]}
                source={
                  Object {
                    "testUri": "../../../src/assets/icono-rombo-titulo.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 0,
                      "width": 50,
                    },
                    undefined,
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 3,
                  "left": 30,
                  "position": "absolute",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#141414",
                    "fontSize": 20,
                  }
                }
              >
                DATOS PROFESIONALES
                <Text
                  style={
                    Object {
                      "fontWeight": "800",
                    }
                  }
                />
              </Text>
            </View>
          </View>
          <View>
            <View>
              <TextInput
                maxLength={20}
                onChange={[Function]}
                placeholder="Nombre Estudio (opcional)"
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="studioName"
                value=""
              />
            </View>
            <View>
              <TextInput
                maxLength={20}
                onChange={[Function]}
                placeholder="Nombre profesional *"
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="professionalName"
                value=""
              />
            </View>
            <View>
              <TextInput
                keyboardType="numeric"
                maxLength={20}
                onChange={[Function]}
                placeholder="Telefono de contacto *"
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="phone"
                value=""
              />
            </View>
            <View>
              <TextInput
                keyboardType="email-address"
                maxLength={20}
                onChange={[Function]}
                placeholder="Email de contacto *"
                style={
                  Object {
                    "borderBottomColor": "#141414",
                    "borderBottomWidth": 1,
                    "padding": 20,
                  }
                }
                testID="contactEmail"
                value=""
              />
            </View>
          </View>
          <View>
            <TextInput
              maxLength={20}
              onChange={[Function]}
              placeholder="Horario *"
              style={
                Object {
                  "borderBottomColor": "#141414",
                  "borderBottomWidth": 1,
                  "padding": 20,
                }
              }
              testID="openingHours"
              value=""
            />
          </View>
          <View>
            <TextInput
              maxLength={20}
              onChange={[Function]}
              placeholder="Dirección *"
              style={
                Object {
                  "borderBottomColor": "#141414",
                  "borderBottomWidth": 1,
                  "padding": 20,
                }
              }
              testID="direction"
              value=""
            />
          </View>
          <View>
            <TextInput
              maxLength={20}
              onChange={[Function]}
              placeholder="Estilos *"
              style={
                Object {
                  "borderBottomColor": "#141414",
                  "borderBottomWidth": 1,
                  "padding": 20,
                }
              }
              testID="tattooStyles"
              value=""
            />
          </View>
        </View>
        <View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": false,
              }
            }
            accessible={true}
            pointerEvents="auto"
          >
            <RNCCheckbox
              boxType="square"
              forwardedRef={null}
              onCheckColor="#7C6B53"
              onTintColor="#7C6B53"
              onValueChange={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "height": 32,
                    "width": 32,
                  },
                  undefined,
                ]
              }
              testID="colaboration"
              tintColor="#141414"
              value={false}
            />
          </View>
          <Text
            style={
              Object {
                "fontSize": 18,
                "textAlign": "center",
              }
            }
          >
            Colaboración con otros estudios / tatuadores?
          </Text>
          <View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderColor": "black",
                  "borderWidth": 1,
                  "marginBottom": 20,
                  "marginTop": 20,
                  "opacity": 1,
                  "padding": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "fontWeight": "900",
                    "textAlign": "center",
                  }
                }
              >
                REGISTRARME COMO TATUADOR
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
